### PR TITLE
Changes implicit_nulls from True to False by default

### PR DESCRIPTION
**Pull request by lapointexavier**
https://app.asana.com/0/13349077888747/134423411674325
### Question
- Took the lazy approach in the tests by changing the kwarg where needed, I guess the coverage is still good, even though the original testing approach was more influenced by implicit_nulls=True?
- Might want to fix the `strict` bug at the same time in this PR ...

### CODE REVIEW

**Comment by benauthor** [comment]
We can do the strict bug in a second PR and just cut one release when both are merged.

File: tests/test_bfh.py [hunk 1]
```diff
@@ -172,9 +172,9 @@ def test_can_init_arrays_with_dict(self):
 
     def test_implicit_nulls(self):
         """Implicit nulls by default"""
```
Comment (benauthor) [inline tests/test_bfh.py]: this comment is now a lie

File: tests/test_bfh.py [hunk 2]
```diff
@@ -172,9 +172,9 @@ def test_can_init_arrays_with_dict(self):
...
-        result = Conversation().serialize()
+        result = Conversation().serialize(implicit_nulls=True)
         self.assertEqual({}, result)
-        result = Conversation(numbers=[1, 2]).serialize()
+        result = Conversation(numbers=[1, 2]).serialize(implicit_nulls=True)
         self.assertEqual({"numbers": [1, 2]}, result)
         result = Conversation().serialize(implicit_nulls=False)
```
Comment (benauthor) [inline tests/test_bfh.py:183]: can you add a separate test that illustrates that the default behavior is now False?

File: tests/test_fields.py [hunk 3]
```diff
@@ -378,7 +378,7 @@ def test_empty_fields_serialized_with_defaults(self):
         second_schema = self.set_up_schema(None)
         second_source = second_schema()
 
-        self.assertEqual({}, second_source.serialize())
+        self.assertEqual({}, second_source.serialize(implicit_nulls=True))
```
Comment (benauthor) [inline tests/test_fields.py]: this whole test has an outdated premise "empty fields serialized with defaults", can probably just remove it. or rewrite to communicate that the presence or absence of a default doesn't have any special behavior beyond the ordinary implicit_nulls behavior

File: tests/test_bfh.py (hunk 2 above)
Comment (lapointexavier) [inline tests/test_bfh.py:183]: Sure thing

File: tests/test_transformations.py [hunk 4]
```diff
@@ -65,7 +65,7 @@ class Myschema(Schema):
         m2 = Myschema()
 
         self.assertEqual(first, All()(m).serialize())
-        self.assertEqual({}, All()(m2).serialize())
+        self.assertEqual({}, All()(m2).serialize(implicit_nulls=True))
```
Comment (benauthor) [inline tests/test_transformations.py]: this is one where the test is more meaningful if you change the asserted value and let implicit_nulls=False (implicitly)

File: tests/test_transformations.py [hunk 5]
```diff
@@ -462,8 +462,8 @@ class OuterNoschema(Mapping):
             None,
         ]
         for source in empties:
-            self.assertEqual({}, Outer().apply(source).serialize())
-            self.assertEqual({}, OuterNoschema().apply(source).serialize())
+            self.assertEqual({}, Outer().apply(source).serialize(implicit_nulls=True))
```
Comment (benauthor) [inline tests/test_transformations.py:465]: this one makes sense like this 👍

File: tests/test_transformations.py [hunk 6]
```diff
@@ -476,7 +476,7 @@ class Hm(Mapping):
         second = {}
         third = {"wow": 3}
         result1 = Hm().apply(first).serialize()
-        result2 = Hm().apply(second).serialize()
+        result2 = Hm().apply(second).serialize(implicit_nulls=True)
```
Comment (benauthor) [inline tests/test_transformations.py]: could go either way on this, but i think change the assertion instead: `assert result2 == {"wow": None}`

File: bfh/__init__.py [hunk 7]
```diff
@@ -41,7 +41,7 @@ def __init__(self, *args, **kwargs):
         """
         Args:
            Pass a dictionary a single positional argument and it will be
```
Comment (benauthor) [inline bfh/__init__.py]: "Pass a dictionary _as_ a" while we're copy editing

**Comment by benauthor** [comment]
LGTM, maybe add a stanza to the readme about serialization and the implicit_nulls behavior?

**Pull request merged**